Revise Bonsai docs (#184)
risc0/website: docs/bonsai/bonsai-overview.md

Bonsai is still in early development. As such we have concentrated on a few use cases: At the moment, we have the core Bonsai proving service and an ETH Relay for accessing Bonsai from Ethereum. This page will keep you updated on the status of these and other Bonsai tools.

## Architecture Example

![Bonsai ETH Relay overview](../img/eth-relay-diagram.jpg)

<!-- revise diagram with "you create"/"bonsai creates" colors -->

You can use Bonsai from an Ethereum smart contract with the pictured architecture. In this diagram, all you are responsible for as a developer using Bonsai are:
* The smart contract invoking Bonsai
* The smart contract handling Bonsai's results via callback
* The program you want Bonsai to prove

This architecture utilizes both the ETH Relay and the proving service; **other architectures are possible**, and as we develop and publish further tools we'll update with other example architectures.

> ## Harnessing the zkVM

Reviewer: I think it would be worth revising this heading to structurally align with the `OneWordTitle: Short Descriptive Tagline` format of the next two headings to visually signal to readers that these three sections are analogous in intent. 

Perhaps: `RISC Zero zkVM: The proving engine behind Bonsai`

_Status: [Available and open source](https://github.com/risc0/risc0)_

Under the hood, Bonsai uses the RISC Zero zkVM to prove your programs. With the zkVM, you have access to recursive proofs, a general-purpose circuit (with a bespoke circuit compiler), state continuations, and continuous improvements to the proving algorithm.

## Bonsai: A zero-knowledge proving service

_Status: Pre-Alpha, [waitlist] available_

> The Bonsai proving service can be used standalone or directly integrated with smart contracts and blockchains. If you're interested in Bonsai, join our [waitlist] and check back here for more news.

Reviewer: This feels quite bare; I'd suggest adding one or two links to learn more here. 

Some options include the litepaper, one of Jeremy's Bonsai talk in Denver, Brian's talk in Lisbon, or this excerpt from my talk at ZK symposium: https://youtu.be/nVAs2i-_Iyo?t=3044

## Bonsai ETH Relay: Connecting off-chain proofs and on-chain applications

_Status: Pre-Release, [documentation](eth-relay/eth-relay-overview.md) and [starter template](https://github.com/risc0/bonsai-starter-template) available_

> The Bonsai ETH Relay connects your Ethereum dApps to Bonsai. Smart contract developers can call our proving service through an on-chain relay contract connected to the Bonsai proving service. If you'd prefer to engage with Bonsai proving service directly, we'll be rolling out the means to do so, but we're excited to offer this option.

Reviewer: ```suggestion
The Bonsai ETH Relay connects your Ethereum dApps to Bonsai. Smart contract developers can call our proving service through an on-chain relay contract connected to the Bonsai proving service. If you'd prefer to engage with Bonsai proving service directly, we'll be rolling out the means to do so in the near future.
```

## Where we're headed

> We're building technology that allows anyone to generate highly performant zero-knowledge proofs for a variety of applications. We want to make zero-knowledge proof integration into Ethereum, L1 blockchains, Cosmos app chains, L2 rollups, and dApps possible in a matter of days with minimal development effort. To learn more, read our [litepaper].

Reviewer: Days seems rather tame of an ambition, doesn't it? Hours seems reasonable in the near future and minutes or seconds feels like an appropriate ambition here...

```suggestion
We're building technology that allows anyone to generate highly performant zero-knowledge proofs for a variety of applications. We want to make zero-knowledge proof integration into Ethereum, L1 blockchains, Cosmos app chains, L2 rollups, and dApps possible with minimal development effort. To learn more, read our [litepaper].
```
